Block/Account/CompanyData.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Devall\Company\Block\Account;
+
+use Magento\Framework\View\Element\Template;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Model\Session as CustomerSession;
+use Devall\Company\Model\CompanyRepository;
+use Devall\Company\Model\ResourceModel\Company\CollectionFactory;
+use Devall\Company\Model\Company as CompanyModel;
+
+class CompanyData extends Template {
+
+    const REST_API_URL = '/rest/V1/devall_company';
+
+    /**
+     * @var Devall\Company\Model\ResourceModel\Company\Collection
+     */
+    private $collection;
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepositoryInterface;
+    /**
+     * @var CompanyRepository
+     */
+    private $companyRepository;
+    /**
+     * @var CustomerSession
+     */
+    private $customerSession;
+
+    /**
+     * CompanyData constructor.
+     * @param Template\Context $context
+     * @param Collection $collection
+     * @param CompanyRepository $companyRepository
+     * @param CustomerSession $customerSession
+     * @param CustomerRepositoryInterface $customerRepositoryInterface
+     * @param array $data
+     */
+    public function __construct(
+        Template\Context $context,
+        CollectionFactory $collectionFactory,
+        CompanyRepository $companyRepository,
+        CustomerSession $customerSession,
+        CustomerRepositoryInterface $customerRepositoryInterface,
+        array $data = []
+    ) {
+        $this->companyRepository = $companyRepository;
+        $this->customerSession = $customerSession;
+        $this->collection = $collectionFactory->create();
```

Comment:
Generally dont want to call any functions during construct except parent construct. Initialize the factory here, and then do collectionFactory->create() in your getCompanyCollection function.